lib/datasource/terraform-provider/index.ts

```diff
@@ -63,9 +63,10 @@ export class TerraformProviderDatasource extends Datasource {
     registryURL: string,
     repository: string
   ): Promise<ReleaseResult> {
-    const serviceDiscovery = await getTerraformServiceDiscoveryResult(
-      registryURL
-    );
+    const serviceDiscovery =
+      await new TerraformModuleDatasource().getTerraformServiceDiscoveryResult(
```

Comment:
😕  Can we create a base class and extend instead of doing this? Or simply extend `TerraformModuleDatasource` ?

Reply:
I'll play around and see which way comes out cleaner.

Reply:
I extracted an abstract `TerraformDatasource` which is currently located in `terraform-module/base.ts`. I'd like to move it to a `terraform` directory, but there'd be more tests that need modification, so I think it's okay for now.